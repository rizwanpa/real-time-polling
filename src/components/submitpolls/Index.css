.survey-index{
    width: 100%;
    height: 100%;
    padding: 0px;
    margin: 0px;
    display: flex;
    justify-content: center;
    align-items: center;
    background: #FFFFFF;
}
.survey-card{
  width: 100%;
  height: 100%;
  padding: 10px;
  /* margin: 5px; */
  display: flex;
  justify-content: center;
  background: #fffcf1;
}
.survey-box{
    width: 400px;
    height: 300px;
    border-radius: 10px;
    border: 1px solid #ccc;
    background: #fbf9f6;
    display: flex;
    flex-direction: column;
}
.survey-poll {
    width: 70%;
}
.survey-err {
  /* width: 400px; */
  margin: 1em 10%;
  padding: 10px;
  border: 1px solid #cc3300;
  background: #ffece6;
  border-radius: 10px;
  color: #cc3300;
  font-weight: 500;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.list-group{
    margin-bottom: 0px;
    padding-left: 0px;
}
.list-group-item {
    position: relative;
    display: block;
    padding: 10px 15px;
    margin-bottom: -1px;
    background-color: #fff;
    border: 1px solid #ddd;
}
div.survey-header {
  text-align: center;
  padding: 5px 0;
}
div.survey-code {
  display: flex;
  justify-content: center;
  align-items: center;
  height: calc(100% - 42px);
}
input.code-input {
  height: 20%;
  width: 80%;
  text-align: center;
  font-size: 30px;
  padding: 25px;
  /* text-transform: uppercase; */
}
div.submit-code {
  padding: 5px 0;
  text-align: center;
  letter-spacing: 2px;
}
div.submit-code button {
  border-radius: 5px;
}
input.code-input::placeholder {
  font-weight: 200;
  color: rgba(0,0,0,0.3);
  text-transform: lowercase;
}
.width-100-per {
  width: 100%;
}