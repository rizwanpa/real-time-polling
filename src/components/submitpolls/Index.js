import React, { Component, createRef } from "react";
import { connect } from 'react-redux';
import {
  Form,
  Input,
  Button,
  Card,
  Radio,
  Checkbox
} from "antd";
import { DownloadOutlined } from '@ant-design/icons';

import { getPollByUuid } from "./../../actions/submitPoll";
import "./Index.css";

class Index extends Component {
  constructor(props) {
    super(props);
    this.state = {
      surveyCode: ''
    };
  }
  formRef = createRef();
  formUuidRef = createRef();
  formPollRef = createRef();
  codeInputRef = createRef();
  componentDidMount() {
    console.log('index==> componentDidMount', this.props);
  }
  componentWillReceiveProps(nextProps) {
    if (!this.props.match.params.uuid) {
      let uuid = nextProps.poll[0].uuid;
      this.props.history.push(`/submit-poll/${uuid}`)
    }
  }
  componentDidUpdate(prevProps, prevState) {
   
  }
  onFinish = formData => {
    
    this.props.getPollByUuidAction(formData.uuid);
  }

  change

  render() {
    return (
      <div className="survey-index">
        {(!this.props.match.params.uuid) ?
          <div className="survey-box">
            {/* <Form
              
              layout="vertical"
              size="small"
              onFinish={this.onFinish}
              ref={this.formRef}
            >
              <Form.Item
                label="Poll Code"
                validateTrigger={["onChange", "onBlur"]}
                rules={[
                  {
                    required: true,
                    whitespace: false,
                    message: "Please enter poll code to continue."
                  }
                ]}
                name="uuid"
              >
                <Input
                  ref={this.formUuidRef}
                  autoComplete="off"
                  placeholder="Enter 6 digit poll code"
                />
              </Form.Item>
              <Button type="primary" htmlType="submit">
                Submit Poll
            </Button>
            </Form> */}
            <div className="survey-header">
              Survey Code
            </div>
            <div className="survey-code">
              <input className="code-input"  placeholder="code"  onChange={(e) => this.setState({surveyCode:e.target.value})}/>
            </div>
            <div className="submit-code">
              <Button type="primary" shape="round" size={200} disabled={!this.state.surveyCode.length}>
                START
              </Button>
            </div>
          </div>
          :
          <div className='survey-poll'>
            <Form
              labelCol={{ span: 14 }}
              wrapperCol={{ span: 14 }}
              layout="vertical"
              size="small"
              onFinish={this.onFinish}
              ref={this.formPollRef}
            >
            {this.props.poll && 
            this.props.poll.map((poll)=>{
              return (
              <Card key = {poll.id} title={poll.title}>
                {poll.questions && poll.questions.map((question)=>{
                  return (<Card
                    key={question.id}
                    style={{ marginTop: 16 }}
                    type="inner"
                    title={question.question}
                    >
                      <div >
                      <ul className="list-group">
                    {
                      question.options && question.options.map((option)=>{
                        let typeOption = question.type ? <Checkbox>{option.option}</Checkbox>: <Radio>{option.option}</Radio>
                      return (
                        <li key={option.option} className="list-group-item">{typeOption}</li>
                        
                        )
                      })
                    }
                    </ul>                      
                    </div>
                  </Card>)

                })}
              </Card>)

            })
                }
            </Form>  
          </div>
      }
      </div>
    );
  }
}

const mapStateToProps = state => {
  console.log('~~~~~~~~~~~~~submit-polls state--------', state);
  return { poll: state.pollForVote.poll };
};

const mapDispatchToProps = {
  getPollByUuidAction: getPollByUuid
};

export default connect(mapStateToProps, mapDispatchToProps)(Index);
