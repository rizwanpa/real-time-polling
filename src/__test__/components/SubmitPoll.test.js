import React from "react";
import { Provider } from "react-redux";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import configureStore from "redux-mock-store"; // Smart components

// Component to be tested
import Index from "./../../components/submitpolls/Index";
configure({ adapter: new Adapter() });
// create any initial state needed
const initialState = {};
// here it is possible to pass in any middleware if needed into //configureStore
const mockStore = configureStore();
const store = mockStore();

describe("Index Component", () => {
  let wrapperIndex;
  // our mock Index function to replace the one provided by mapDispatchToProps
  const mockIndexfn = jest.fn();
  describe("render()", () => {
    test("renders the component", () => {
      const wrapper = shallow(
        <Provider store={store}>
          <Index />
        </Provider>
      );
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
