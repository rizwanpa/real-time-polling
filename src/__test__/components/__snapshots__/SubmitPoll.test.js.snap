// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Component render() renders the component 1`] = `<Connect(Index) />`;
